src/libfetchers/git-utils.cc

```diff
@@ -205,6 +205,24 @@ static git_packbuilder_progress PACKBUILDER_PROGRESS_CHECK_INTERRUPT = &packBuil
 
 } // extern "C"
 
+static void initRepoAtomically(std::filesystem::path &path, bool bare) {
```

Comment:
```suggestion
static void initRepoAtomically(std::filesystem::path & path, bool bare) 
{
    if (pathExists(path.string())) return;
```
to make this function idempotent.